lib/formats/bs_format_ripejson.c

```diff
@@ -220,6 +226,46 @@ static int process_common_fields(bgpstream_format_t *format,
   return 0;
 }
 
+// readline
+static int readline(bgpstream_format_t *format)
+{
+  int total_read = 0;
+  int current_read = 0;
+  char *buffer_ptr = STATE->json_string_buffer;
+
+  while(1){
+    // read line, including linebreak
+    current_read = bgpstream_transport_readline(
+      format->transport, buffer_ptr, BUFSIZE - total_read, 0);
+    total_read+=current_read;
+
+    if(current_read == 0){
+      // reach EOF, no bytes read, return 0
+      return 0;
+    }
+
+    if(current_read < BUFSIZE - total_read -1 ||
+      buffer_ptr[current_read-1] == '\n'){
+      // if read less than buffer, or last byte is an newline, readline is finished
+
+      // replace linebreak with null
+      buffer_ptr[current_read-1] = '\0';
+      buffer_ptr += current_read - 1;
+      break;
+    }
+
+    // if reaches here: no newline received yet, need to read more
+    // reallocate memory
+    BUFSIZE += JSON_BUFFER_BLOCK_SIZE;
+    STATE->json_string_buffer = (char*) realloc (STATE->json_string_buffer, BUFSIZE * sizeof(char));
+    buffer_ptr = &STATE->json_string_buffer[total_read - 1];
+    bgpstream_log(BGPSTREAM_LOG_INFO, "Json string buffer size increased to %d", BUFSIZE);
+  }
+
+  // calcualte how many bytes actually read
+  return buffer_ptr - STATE->json_string_buffer;
```

Comment:
shouldn't this be `total_read`?

Reply:
fixed in e0f2603c68b265df27ba39014439d753f8ceaeb7